src/datamodel/serializer/Uuid.cpp

```diff
@@ -10,3 +10,122 @@
+	       ((rand_b >> 48) & 0x3FFF))  // Variant and randb
+	   << "-" << std::setw(12) << (rand_b & 0xFFFFFFFFFFFF);  // Random number
+
+	return std::move(ss).str();
+}
+
+uprotocol::v1::UUID AsString::deserialize(const std::string& str) {
+	// Check if the UUID string is in the correct format
+	if (str.length() != 36 || str[8] != '-' || str[13] != '-' ||
+	    str[18] != '-' || str[23] != '-') {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
+
+	uprotocol::v1::UUID uuid;
+	uint64_t unix_ts_ms = 0;
+	uint8_t ver = 0;
+	uint16_t counter = 0;
+	uint8_t var = 0;
+	uint64_t rand_b = 0;
+
+	try {
+		// Extract the parts from the UUID string
+		unix_ts_ms = std::stoull(str.substr(0, 8), nullptr, 16) << 16;
+		unix_ts_ms |= std::stoull(str.substr(9, 4), nullptr, 16);
+
+		uint16_t ver_counter = std::stoul(str.substr(14, 4), nullptr, 16);
+		ver = (ver_counter >> 12) & 0xF;
+		counter = ver_counter & 0xFFF;
+
+		uint16_t var_randb = std::stoul(str.substr(19, 4), nullptr, 16);
+		var = (var_randb >> 14) & 0x3;
+		rand_b = static_cast<uint64_t>(var_randb & 0x3FFF) << 48;
+		rand_b |= std::stoull(str.substr(24), nullptr, 16);
+	} catch (const std::exception& e) {
+		throw std::invalid_argument("Invalid UUID string format");
+	}
+
+	// Reconstruct the UUID
+	uuid.set_msb((unix_ts_ms << 16) | (ver << 12) | counter);
+	uuid.set_lsb((static_cast<uint64_t>(var) << 62) | rand_b);
+
+	return uuid;
+}
+
+// Serialization function
+std::vector<uint8_t> AsBytes::serialize(const v1::UUID uuid) {
+	std::vector<uint8_t> bytes(16);
+
+	uint32_t msb_high = htonl(static_cast<uint32_t>(uuid.msb() >> 32));
+	uint32_t msb_low = htonl(static_cast<uint32_t>(uuid.msb() & 0xFFFFFFFF));
+	uint32_t lsb_high = htonl(static_cast<uint32_t>(uuid.lsb() >> 32));
+	uint32_t lsb_low = htonl(static_cast<uint32_t>(uuid.lsb() & 0xFFFFFFFF));
+
+	std::memcpy(bytes.data(), &msb_high, sizeof(uint32_t));
+	std::memcpy(bytes.data() + sizeof(uint32_t), &msb_low, sizeof(uint32_t));
+	std::memcpy(bytes.data() + 2 * sizeof(uint32_t), &lsb_high,
+	            sizeof(uint32_t));
+	std::memcpy(bytes.data() + 3 * sizeof(uint32_t), &lsb_low,
+	            sizeof(uint32_t));
+
+	return bytes;
+}
+
+v1::UUID AsBytes::deserialize(const std::vector<uint8_t>& bytes) {
+	if (bytes.size() != 16) {
+		throw std::invalid_argument("Invalid UUID byte array size");
+	}
+
+	uint32_t msb_high, msb_low, lsb_high, lsb_low;
+
+	std::memcpy(&msb_high, bytes.data(), sizeof(uint32_t));
+	std::memcpy(&msb_low, bytes.data() + sizeof(uint32_t), sizeof(uint32_t));
+	std::memcpy(&lsb_high, bytes.data() + 2 * sizeof(uint32_t),
+	            sizeof(uint32_t));
+	std::memcpy(&lsb_low, bytes.data() + 3 * sizeof(uint32_t),
+	            sizeof(uint32_t));
+
+	uint64_t msb =
+	    (static_cast<uint64_t>(ntohl(msb_high)) << 32) | ntohl(msb_low);
+	uint64_t lsb =
+	    (static_cast<uint64_t>(ntohl(lsb_high)) << 32) | ntohl(lsb_low);
+
+	v1::UUID uuid;
+	uuid.set_msb(msb);
+	uuid.set_lsb(lsb);
+
+	return uuid;
+}
```

Comment:
The more I think about it, the more I start to think we might not need this at all. Converting to/from a string makes a lot of sense for debugging or as a developer interface for loading data, but I'm having trouble coming up with an actual use case for the byte serialization.

I know there was a reason I put this in here back when I did the API, but that might have been a remnant of the way the old API worked.

We can keep it here for now and consider removing it later if there's no need for it.